fendermint/vm/topdown/src/finality/null.rs

```diff
@@ -71,6 +82,17 @@ impl FinalityWithNull {
         height: BlockHeight,
         maybe_payload: Option<ParentViewPayload>,
     ) -> StmResult<(), Error> {
+        if let Some(p) = self.last_committed_finality.read_clone()? {
```

Comment:
```suggestion
        if let Some(p) = self.last_committed_finality.read()? {
```
Here too, the value isn't returned, so you can access it through the `Arc`